root/opt/puppetlabs/puppet/lib/ruby/2.1.0/facter/systemd.rb

```diff
@@ -22,9 +22,15 @@
     confine osfamily: 'RedHat'
     setcode do
         systemd = {
-            "restart" => {}
+            "restart" => {},
+            "error" => ""
         }
-        tmp = Facter::Core::Execution.execute("zgrep ', status=' /var/log/messages* | awk '{print $6,$11}' | sort | uniq -c", :timeout => 60)
+        begin
+            tmp = Facter::Core::Execution.execute("find /var/log/ -maxdepth 1 -name 'messages*' -newermt $(date -d '1 months ago' +@%s) | xargs -- nice zgrep ', status=' |  awk '{print $6,$11}' | sort | uniq -c", :timeout => 60)
```

Comment:
I do not get why the command is invoked using nice: IIRC nice with `-n` argument sets the default niceness.

Reply:
No it sets the default "10" that means a priority below the normal which is "0"

```text
[davidep@runner ~]$ /bin/nice ps -o nice,cmd
 NI CMD
  0 /bin/bash
 10 ps -o nice,cmd
```